packages/database/data-migrations/20240612115151_adds_product_config/data-migration.ts

```diff
@@ -0,0 +1,100 @@
+import { PrismaClient, SurveyType } from "@prisma/client";
+
+const prisma = new PrismaClient();
+
+type channelType = Exclude<SurveyType, "web"> | null;
+
+const main = async () => {
+  await prisma.$transaction(
+    async (tx) => {
+      const startTime = Date.now();
+      console.log("Starting data migration...");
+
+      // Fetch all products
+      const products = await tx.product.findMany({
+        include: {
+          environments: {
+            select: {
+              surveys: {
+                select: {
+                  type: true,
+                },
+              },
+            },
+          },
+        },
+      });
+
+      console.log(`Found ${products.length} products to migrate...\n`);
+
+      const channelStatusCounts = {
+        [SurveyType.app]: 0,
+        [SurveyType.link]: 0,
+        [SurveyType.website]: 0,
+        null: 0,
+      };
+
+      for (const product of products) {
+        const surveyTypes = new Set<SurveyType>();
+
+        // Collect all unique survey types for the product
+        for (const environment of product.environments) {
+          for (const survey of environment.surveys) {
+            surveyTypes.add(survey.type);
+          }
+        }
+
+        let channel: channelType = null;
+
+        if (surveyTypes.size === 0 || surveyTypes.size === 3) {
+          channel = null;
+        } else if (surveyTypes.size === 1) {
+          const type = Array.from(surveyTypes)[0];
+          if (type === SurveyType.web) {
+            channel = null;
+          } else {
+            channel = type;
+          }
+        } else if (surveyTypes.has(SurveyType.link) && surveyTypes.has(SurveyType.app)) {
+          channel = SurveyType.app;
+        } else if (surveyTypes.has(SurveyType.link) && surveyTypes.has(SurveyType.website)) {
+          channel = SurveyType.website;
+        }
```

Comment:
Do we have an unhandled case where we have `app` & `website` surveys inside the product? It's `null` since the initial value of `channel` is `null`, but maybe it makes sense to mention this explicitly like we did with the other cases.

Reply:
In cases like these also comments make sense to make the conditions easier to understand

Reply:
yeah, added the `else if` to have an explicit check for the condition.